9.2/run-postgresql.sh

```diff
@@ -0,0 +1,83 @@
+#!/bin/bash -e
+
+# Be paranoid and stricter than we should be.
+psql_identifier_regex='^[a-zA-Z_][a-zA-Z0-9_]*$'
+psql_password_regex='^[a-zA-Z0-9_~!@#$%^&*()-=<>,.?;:|]+$'
+
+function usage {
+	echo "You must specify following environment variables:"
+	echo "  \$POSTGRESQL_USER (regex: '$psql_identifier_regex')"
+	echo "  \$POSTGRESQL_PASSWORD (regex: '$psql_password_regex')"
+	echo "  \$POSTGRESQL_DATABASE (regex: '$psql_identifier_regex')"
```

Comment:
I'd rather see a failure when that name isn't specified. It's required the
same way username and password is. This is the behavior I'd vote for all
images.

On 6:27pm, Thu, Mar 19, 2015 Jakub Hadvig <notifications@github.com> wrote:

> In 9.2/run-postgresql.sh
> <https://github.com/openshift/postgresql/pull/1#discussion_r26777138>:
>
> > @@ -0,0 +1,83 @@
> > +#!/bin/bash -e
> > +
> > +# Be paranoid and stricter than we should be.
> > +psql_identifier_regex='^[a-zA-Z_][a-zA-Z0-9_]*$'
> > +psql_password_regex='^[a-zA-Z0-9_~!@#$%^&*()-=<>,.?;:|]+$'
> > +
> > +function usage {
> > +	echo "You must specify following environment variables:"
> > +	echo "  \$POSTGRESQL_USER (regex: '$psql_identifier_regex')"
> > +	echo "  \$POSTGRESQL_PASSWORD (regex: '$psql_password_regex')"
> > +	echo "  \$POSTGRESQL_DATABASE (regex: '$psql_identifier_regex')"
>
> In MongoDB I am creating the production database if none is defined. I
> think some similar approach we had in v2. Wouldn't it be better to have it
> in all DB images (also MySQL) ?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/postgresql/pull/1/files#r26777138>.
>